app/src/main/java/org/dicekeys/app/fragments/AssembleFragment.kt

```diff
@@ -1,19 +1,220 @@
 package org.dicekeys.app.fragments
 
+import android.animation.ObjectAnimator
+import android.animation.ValueAnimator
 import android.os.Bundle
+import android.view.LayoutInflater
 import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.ImageView
+import androidx.annotation.LayoutRes
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentStatePagerAdapter
+import androidx.fragment.app.viewModels
+import androidx.navigation.NavOptions
+import androidx.navigation.NavOptionsBuilder
+import androidx.navigation.fragment.findNavController
+import androidx.viewpager.widget.ViewPager
 import dagger.hilt.android.AndroidEntryPoint
 import org.dicekeys.app.AppFragment
+import org.dicekeys.app.NavGraphDirections
 import org.dicekeys.app.R
-import org.dicekeys.app.databinding.AssembleFragmentBinding
-import org.dicekeys.app.viewmodels.DiceKeyViewModel
+import org.dicekeys.app.databinding.*
+import org.dicekeys.app.extensions.clearNavigationResult
+import org.dicekeys.app.extensions.getNavigationResult
+import org.dicekeys.app.fragments.backup.BackupFragment
+import org.dicekeys.app.repositories.DiceKeyRepository
+import org.dicekeys.app.viewmodels.AssembleViewModel
+import org.dicekeys.app.views.DiceKeyView
+import org.dicekeys.app.views.StickerTargetSheetView
+import org.dicekeys.dicekey.DiceKey
+import org.dicekeys.dicekey.Face
+import org.dicekeys.dicekey.FaceRead
+import javax.inject.Inject
 
 @AndroidEntryPoint
-class AssembleFragment: AppFragment<AssembleFragmentBinding>(R.layout.assemble_fragment) {
+class AssembleFragment : AppFragment<AssembleFragmentBinding>(R.layout.assemble_fragment) {
 
+    @Inject
+    lateinit var diceKeyRepository: DiceKeyRepository
+
+    private lateinit var pagerAdapter: AssemblePagerAdapter
+
+
+    val viewModel: AssembleViewModel by viewModels()
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        binding.vm = viewModel
+
+        getNavigationResult<String>(ScanFragment.READ_DICEKEY)?.observe(viewLifecycleOwner) {
+            it?.let {
```

Comment:
stylistically, I find the use of "it" can be confusing when it's not obvious to the reader what the "it" is.  In this case, I'm not sure from context what is being observed, but that may just be my ignorance of this programming style.